tcadmin/resources/client.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+
+import attr
+import datetime
+
+from .resources import Resource
+from .util import description_converter, scopes_converter, list_formatter
+from ..util.scopes import normalizeScopes
+
+FOREVER = datetime.datetime(3000, 1, 1, 0, 0, 0)
+
+
+@attr.s
+class Client(Resource):
+    clientId = attr.ib(type=str)
+    description = attr.ib(type=str, converter=description_converter)
+    scopes = attr.ib(
+        type=tuple, converter=scopes_converter, metadata={"formatter": list_formatter}
+    )
+
+    # NOTE: clients are managed like roles.  Any associated access tokens are
+    # not handled by this library.  Where clients are needed, their access
+    # tokens should be reset manually and the resulting accessToken used as
+    # necessary.  Like roles, the clients managed here last "forever".
```

Comment:
what doesn't last forever?

Reply:
Clients can expire, so this is indicating that tc-admin doesn't let you set the expiration yourself.